starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +82,12 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory inventory = this.mongoTemplate.findById(id, Inventory.class);
+    Optional<Inventory> deletedInv = Optional.ofNullable(inventory);
+    if (deletedInv.isPresent()) {
```

Comment:
Let's make sure the incoming id String is not null or has text, and then we can use Criteria and Query, and a single method on mongoTemplate that will find and delete the desired item. Then we can return that.